prov/rxd/src/rxd_cq.c

```diff
@@ -901,14 +941,31 @@ static void rxd_handle_data(struct rxd_ep *ep, struct rxd_pkt_entry *pkt_entry)
 {
 	struct rxd_data_pkt *pkt = (struct rxd_data_pkt *) (pkt_entry->pkt);
 	struct rxd_x_entry *x_entry;
+	struct dlist_entry *unexp_list;
+	struct rxd_unexp_msg *unexp_msg;
 
 	if (pkt_entry->pkt_size < sizeof(*pkt) + ep->rx_prefix_size) {
 		FI_WARN(&rxd_prov, FI_LOG_CQ,
 			"Cannot process packet smaller than minimum header size\n");
-		return;
+		goto free;
 	}
 
 	if (pkt->base_hdr.seq_no == ep->peers[pkt->base_hdr.peer].rx_seq_no) {
+		ep->peers[pkt->base_hdr.peer].rx_seq_no++;
+		if (pkt->base_hdr.type == RXD_DATA &&
+		    ep->peers[pkt->base_hdr.peer].curr_rx_id ==
+		    (uint16_t) RXD_UNEXP_ID) {
```

Comment:
Is curr_rx_id = 0xFFFF not valid?  Are we guaranteed that this value is only used for unexpected messages?

Reply:
Improbable but I'll add a fix to set the max count of the x_entry pool to avoid that.